### ISSUE TITLE
Add support for simulating EEG data

### ISSUE THREAD

**Issue by ssaket** [unfoldtoolbox/Unfold.jl#43]
Simulate events and EEG data based on 
- StatsBase @formula
- Trials, Times, channels (more flexibility, in terms of choosing covariate)

**Comment by behinger** [comment]
- combining formulas with betas is not straight forward. One always need to specify levels and coding as well. I have simulation code for MixedModelsSim
- Multiple subjects => use MixedModelsSim as a basis and just simulate n=1 with no random effects.
- temporal overlap (results in continuous EEG)
- basis-functions (e.g. hanning)
- several noise functions (white, pink, brown, exponential autocorr, AR1) =>Ask Felix, he implemented those already

=> Have a look at SEREEGA toolbox again, but I think it is too detailed. I.e. we do not need variability in component latencies etc. that is too much imho.

**Comment by ssaket** [comment]
> combining formulas with betas is not straight forward. One always need to specify levels and coding as well. I have simulation code for MixedModelsSim

I agree, it can be complicated and maybe not worth the effort. I had this thought when I was going through the StatsModels package (https://juliastats.org/StatsModels.jl/stable/formula/#The-@formula-language)

**Comment by behinger** [comment]
Also see this: https://github.com/RePsychLing/MixedModelsSim.jl/pull/40

Once I find some time, I will put up a rough sketch for a UnfoldSimulation.jl package with the code that is laying around. I think I have lot's of components already, just need to move them together and combine them

**Comment by behinger** [comment]
Slowly designing some thoughts on `UnfoldSim`



## Design Module
- per eventtype, call MixedModelsSim => one DataFrame per event, + one formula
e.g. `Dict(:eventA => (table,formula), eventB =>(table2,formula2)

This allows the user to get perfect control of any balancing issues.


## Signal Module

- per event: one set of coefficients => Maybe we can also employ MixedModelsSim here
- per event: able to simulate random effect structure subject + item
- stretch-goal: correlation between two coefficients of different events. That is, e.g. continuousB is always larger, when `event B` is following `event A`, but not when it is following `event B`. It is unclear to me how to make the tool flexible enough to allow for such cases.
- 
## Overlap Module
- per-event: Distribution of lags (e.g. uniform, ex-Gaussian) with sensible parameterization
- relation of events (e.g. "AABAABAAB" or "ABABABA" or "AAAABBBBAAAA") needs to be specified. Maybe we can do this already in the Design Module?
- dependence of distribution parameter on events. In some sense we need a second set of formulae + coefficients, just to describe the mean + variability of the duration (it is necessary that we allow for biased  overlap, e.g. more overlap in one condition than another; tricky is, that we dont want to allow any type of overlap (e.g. negative values).

### Noise Module
- default: add noise to continuous signal
- allow noise-structure per event (e.g. linear increase allows for heteroskedasticity), but careful, with overlap, noise-levels adds!

Types: white, pink, brown, exponential,real EEG
